packages/vats/src/orchestration.js

```diff
@@ -0,0 +1,268 @@
+      /**
+       * @type {{
+       *   port: Port;
+       *   connection: Connection | undefined;
+       *   localAddress: string | undefined;
+       *   requestedRemoteAddress: string;
+       *   remoteAddress: string | undefined;
+       *   accountAddress: ChainAddress | undefined;
+       * }}
+       */ (
+        harden({
+          port,
+          connection: undefined,
+          requestedRemoteAddress,
+          remoteAddress: undefined,
+          accountAddress: undefined,
+          localAddress: undefined,
+        })
+      ),
+    {
+      account: {
+        getAccountAddress() {
+          const { accountAddress } = this.state;
+          accountAddress || Fail`account address not available`;
+          return accountAddress;
+        },
+        getLocalAddress() {
+          const { localAddress } = this.state;
+          localAddress || Fail`local address not available`;
+          return localAddress;
+        },
+        getRemoteAddress() {
+          const { remoteAddress } = this.state;
+          remoteAddress || Fail`remote address not available`;
+          return remoteAddress;
+        },
+        getPort() {
+          return this.state.port;
+        },
+        /** @param {Bytes} packetBytes */
+        async executeEncodedTx(packetBytes) {
+          const { connection } = this.state;
+          connection || Fail`connection not available`;
+          // @ts-expect-error Fail does not satisfy undefined
+          return E(connection).send(packetBytes);
+        },
+        /** @param {import('@agoric/vats/src/localchain').Proto3Jsonable[]} _msgs */
+        async executeTx(_msgs) {
+          // XXX encode to protobuf, and call this.executeEncodedTx(...)
+          Fail`executeTx not implemented yet`;
+        },
+        async deposit(_payment) {
+          // XXX deposit an ERTP payment to the remote account
+          Fail`deposit not implemented yet`;
+        },
+        async getPurse(_brand) {
+          Fail`getPurse not implemented yet`;
+        },
+        async close() {
+          const { connection } = this.state;
+          connection || Fail`connection not available`;
+          // XXX retrieve all assets first?
+          // XXX do we also revoke the port?
+          // @ts-expect-error Fail does not satisfy possibly undefined
+          return E(connection).close();
+        },
+      },
+      connectionHandler: {
+        /**
+         * @param {Connection} connection
+         * @param {string} localAddr
+         * @param {string} remoteAddr
+         * @param {ConnectionHandler} _connectionHandler
+         */
+        async onOpen(connection, localAddr, remoteAddr, _connectionHandler) {
+          trace(`ICA Channel Opened for ${localAddr} at ${remoteAddr}`);
+          this.state.connection = connection;
+          this.state.remoteAddress = remoteAddr;
+          this.state.localAddress = localAddr;
+          // XXX parseAddress currently throws, should it return '' instead?
```

Comment:
I'd expect a parser to throw/error if I give it something that doesn't conform.  but we might want to provide a utility like `isValid` that the caller can use to handle problems before causing a throw